Content.Server/GameObjects/EntitySystems/PointingSystem.cs

```diff
@@ -0,0 +1,156 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using Content.Server.GameObjects.Components.Pointing;
+using Content.Shared.Input;
+using Content.Shared.Interfaces;
+using JetBrains.Annotations;
+using Robust.Server.Interfaces.Player;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Input.Binding;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.Interfaces.Timing;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Players;
+
+namespace Content.Server.GameObjects.EntitySystems
+{
+    [UsedImplicitly]
+    public class PointingSystem : EntitySystem
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
+        [Dependency] private readonly ITileDefinitionManager _tileDefinitionManager = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+#pragma warning restore 649
+
+        private static readonly TimeSpan PointDelay = TimeSpan.FromSeconds(0.5f);
+
+        /// <summary>
+        ///     A dictionary of players to the last time that they
+        ///     pointed at something.
+        /// </summary>
+        private Dictionary<ICommonSession, TimeSpan> _pointers = default!;
```

Comment:
Just initialize this in the field instead of doing `= default!;` and then initializing it later.

Reply:
You're not clearing out references from this dictionary meaning that they'll stick here until the server is restarted, even if players disconnect. Hook `OnPlayerStatusChanged` to remove dead sessions on disconnect.